library/string.lisp

```diff
@@ -39,6 +39,13 @@
 ;;; String
 ;;;
 
+(cl:eval-when (:compile-toplevel :load-toplevel)
+  (cl:defmacro define-instance-into-rational-string (type)
+    `(define-instance (Into ,type String)
+       (define (into z)
+         (lisp String (z)
+           (cl:format cl:nil "~D" z))))))
```

Comment:
`~D` won't work for rationals I don't think (??); I forgot.

Reply:
I tested for a simple case for all the types which I implemented with this macro and it seemed to work just fine.

Reply:
CLHS 22.3.2.2 says `~D` "should" have an argument which is an integer.

But it does say that if it's not an integer, it'll fall back on `~A`. So your code should work, but it's just a little strange to see.

Reply:
I changed this to `define-instance-into-integral-string` and I handled fraction separately with `"~A"` now.